Modix.Data/ModixContextAutoMigrationStartupAction.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Modix.Data
+{
+    public interface IModixContextAutoMigrationStartupAction
```

Comment:
I think this should implemented on ScopedStartupActionBase which actually defines WhenDone. I was confused where that was coming from for a bit.

Reply:
That makes it impossible to request this dependency from the DI container.

Reply:
You should move WhenDone to an interface on the class then make this interface implement that one. Then any implementation of ScopedActionStartupBase inherits IWhenDone automatically and you can still resolve IModixContextAutoMigrationStartupAction

Reply:
So, `IModixContextAutoMigrationStartupAction` becomes literally just a tag interface. I suppose I could get behind that. I think what led me here was the idea that maybe not all startup actions NEED to supply a `WhenDone` 'cause there may not be any other actions that need to consume it.